django/forms/fields.py

```diff
@@ -435,9 +436,18 @@ def strptime(self, value, format):
         return datetime.datetime.strptime(value, format).time()
 
 
+class DatetimeFormatsIterator:
+    def __iter__(self):
+        for item in chain(
+            formats.get_format('DATETIME_INPUT_FORMATS'),
+            formats.get_format('DATE_INPUT_FORMATS')
+        ):
+            yield item
```

Comment:
`itertools.chain()` returns an iterator so I think we can simply use it in `DateTimeField.input_formats`, :thinking: :
```python
input_formats = chain(
    formats.get_format('DATETIME_INPUT_FORMATS'),
    formats.get_format('DATE_INPUT_FORMATS'),
)
```
What do you think?

Reply:
But `formats.get_format()` is not lazy, am I right? :thinking:. We can also use `yield from ... ` without `chain()`, e.g.
```python
def __iter__(self):
    yield from formats.get_format('DATETIME_INPUT_FORMATS')
    yield form formats.get_format('DATE_INPUT_FORMATS')
```